### PR TITLE
Remove HeaderDecoder from HttpServer

### PR DESCRIPTION
**Pull request by legionth**
HeaderDecoder isn't needed and only adds more complexity.

### CODE REVIEW

**Comment by legionth** [comment]
Ping @clue.

File: src/HttpServer.php [hunk 1]
```diff
@@ -60,48 +60,84 @@ public function addMiddleware($middleware)
...
+                if ($data !== '') {
+                    $connection->emit('data', array($data));
+                }
+            }
+        };
+
+        $connection->on('data', $listener);
+    }
+
+    public function handleBody($request, $connection)
```
Comment (clue) [inline src/HttpServer.php]: `@internal`? typehints?

**Review by clue — CHANGES_REQUESTED** [review]
Nice! :+1: (+74 −236)

Minor API doc issue only

**Comment by legionth** [comment]
Ping @clue

**Review by clue — APPROVED** [review]
LGTM :shipit:

**Pull request merged**